tests/explainers/test_tree.py

```diff
@@ -1205,13 +1058,175 @@ def test_HistGradientBoostingRegressor(self):
+    def test_xgboost(self):
+        """Test the basic `shap_values` calculation."""
+        xgboost = pytest.importorskip('xgboost')
+
+        # train lightgbm model
+        X, y = shap.datasets.california(n_points=500)
+        model = xgboost.XGBRegressor().fit(X, y)
+
+        # explain the model's predictions using SHAP values
+        ex = shap.TreeExplainer(model)
+        shap_values = ex.shap_values(X)
+        predicted = model.predict(X)
+
+        expected_diff = np.abs(shap_values.sum(1) + ex.expected_value - predicted).max()
+        assert expected_diff < 1e-4, "SHAP values don't sum to model output!"
+
+    def test_xgboost_direct(self, random_seed):
+        xgboost = pytest.importorskip('xgboost')
+
+        rs = np.random.RandomState(random_seed)
+        N = 100
+        M = 4
+        X = rs.standard_normal(size=(N, M))
+        y = rs.standard_normal(size=N)
+
+        model = xgboost.XGBRegressor(random_state=rs)
+        model.fit(X, y)
+
+        explainer = shap.TreeExplainer(model)
+        shap_values = explainer.shap_values(X)
+
+        assert np.allclose(shap_values[0, :], _brute_force_tree_shap(explainer.model, X[0, :]))
+
+    def test_xgboost_multiclass(self):
+        xgboost = pytest.importorskip('xgboost')
+
+        # train XGBoost model
+        X, y = shap.datasets.iris()
+        model = xgboost.XGBClassifier(n_estimators=1, max_depth=4)
+        model.fit(X, y)
+
+        explainer = shap.TreeExplainer(model)
+
+        assert np.allclose(explainer.model.predict(X), model.predict(X, output_margin=True))
+
+        shap_values = explainer.shap_values(X)
+
+        # ensure plot works for first class
+        shap.dependence_plot(0, shap_values[0], X, show=False)
+
+    def test_xgboost_ranking(self):
+        xgboost = pytest.importorskip('xgboost')
+
+        # train xgboost ranker model
+        x_train, y_train, x_test, _, q_train, _ = shap.datasets.rank()
+        params = {
+            "objective": "rank:pairwise",
+            "learning_rate": 0.1,
+            "gamma": 1.0,
+            "min_child_weight": 0.1,
+            "max_depth": 5,
+            "n_estimators": 4,
+        }
+        model = xgboost.sklearn.XGBRanker(**params)
+        model.fit(x_train, y_train, group=q_train.astype(int))
+        _validate_shap_values(model, x_test)
+
+    def test_xgboost_mixed_types(self):
+        xgboost = pytest.importorskip('xgboost')
+
+        X, y = shap.datasets.california(n_points=500)
+        X["HouseAge"] = X["HouseAge"].astype(np.int64)
+        X['IsOld'] = (X['HouseAge'] > 30)
+        bst = xgboost.train({"learning_rate": 0.01}, xgboost.DMatrix(X, label=y), 1000)
+        shap_values = shap.TreeExplainer(bst).shap_values(X)
+        shap.dependence_plot(0, shap_values, X, show=False)
+
+    def test_xgboost_classifier_independent_margin(self, random_seed):
+        xgboost = pytest.importorskip("xgboost")
+        # train XGBoost model
+        rs = np.random.RandomState(random_seed)
```

Comment:
same comment here in `test_xgboost_classifier_independent_margin` regarding the random seed. Please retain the FIXME and remove `random_seed` for now.

Reply:
Commit reverted